recipes/openssl/1.x.x/conanfile.py

```diff
@@ -728,35 +725,31 @@ def build(self):
 
                 self._patch_install_name()
 
-                if self._use_nmake and self._full_version < "1.1.0":
-                    if not self.options.no_asm and self.settings.arch == "x86":
-                        self.run(r"ms\do_nasm")
-                    else:
-                        self.run(r"ms\do_ms" if self.settings.arch == "x86" else r"ms\do_win64a")
-
-                    self._replace_runtime_in_file(os.path.join("ms", "nt.mak"))
-                    self._replace_runtime_in_file(os.path.join("ms", "ntdll.mak"))
-                    if self.settings.arch == "x86":
-                        replace_in_file(self, os.path.join("ms", "nt.mak"), "-WX", "")
-                        replace_in_file(self, os.path.join("ms", "ntdll.mak"), "-WX", "")
-
-                    # NMAKE interprets trailing backslash as line continuation
-                    replace_in_file(self, self._nmake_makefile, 'INSTALLTOP=\\', 'INSTALLTOP=/')
-                        
-                    autotools.make(args=[f'-f {self._nmake_makefile}'])
-                else:
+                if not self._use_nmake:
                     autotools.make()
+                else:
+                    if self._full_version >= "1.1.0":
+                        self.run(f'nmake /F Makefile')
+                    else: # nmake 1.0.2 support
+                        # Note: 1.0.2 should not be used according to the OpenSSL Project
+                        #       See https://www.openssl.org/source/
+
+                        if not self.options.no_asm and self.settings.arch == "x86":
+                            self.run(r"ms\do_nasm")
+                        else:
+                            self.run(r"ms\do_ms" if self.settings.arch == "x86" else r"ms\do_win64a")
+
+                        self._replace_runtime_in_file(os.path.join("ms", "nt.mak"))
+                        self._replace_runtime_in_file(os.path.join("ms", "ntdll.mak"))
+                        if self.settings.arch == "x86":
+                            replace_in_file(self, os.path.join("ms", "nt.mak"), "-WX", "")
+                            replace_in_file(self, os.path.join("ms", "ntdll.mak"), "-WX", "")
+
+                        # NMAKE interprets trailing backslash as line continuation
+                        replace_in_file(self, self._nmake_makefile, 'INSTALLTOP=\\', 'INSTALLTOP=/')
+
+                        self.run(f'nmake /F {self._nmake_makefile}')
   
-    @property
-    def _make_program(self):
-        if self._use_nmake:
-            return "nmake"
-        make_program = os.getenv("CONAN_MAKE_PROGRAM") or shutil.which("make") or shutil.which('mingw32-make')
```

Comment:
👍  for future readers, this is replaced by the conf `tools.gnu:make_program` https://docs.conan.io/en/latest/reference/config_files/global_conf.html?highlight=make%20program#tools-configurations with the new generators